workers/task-driver/src/state_migration/remove_old_proofs.rs

```diff
@@ -0,0 +1,67 @@
+//! Removes old proofs from the state
+
+use common::types::wallet::OrderIdentifier;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use state::{State, storage::tx::RwTxn};
+use tracing::{info, warn};
+
+/// Removes old proofs from the state
+pub async fn remove_old_proofs(state: &State) -> Result<(), String> {
+    state
+        .with_write_tx(move |tx| {
+            if let Err(e) = remove_old_proofs_with_tx(tx) {
+                warn!("error removing old proofs: {e}");
+            }
+
+            Ok(())
+        })
+        .await?;
+
+    Ok(())
+}
+
+/// Remove old proofs with a tx
+fn remove_old_proofs_with_tx(tx: &RwTxn) -> Result<(), String> {
+    let cursor = tx.inner().cursor::<String, AnyValue>("proofs").map_err(|e| e.to_string())?;
+    let iter = cursor.into_iter();
+
+    let mut n_deleted = 0;
+    for res in iter {
+        let (k, _v) = res.map_err(|e| e.to_string())?;
+        let oid = k.split(":").last().unwrap();
+        let oid = OrderIdentifier::parse_str(oid).map_err(|e| e.to_string())?;
+
+        if tx.contains_order(&oid).map_err(|e| e.to_string())? {
+            continue;
+        }
+
+        tx.delete_proofs_for_order(&oid).map_err(|e| e.to_string())?;
+        n_deleted += 1;
+    }
+
+    info!("deleted {n_deleted} old proofs");
```

Comment:
nit: would be more accurate to say `deleted old proofs for {n_deleted} cancelled orders`